### ISSUE TITLE
[Dev Info] Battle.net Launcher doesn't start starting wine-staging at commit 02913f7

### ISSUE THREAD

**Issue by AdelKS** [lutris/wine#30]
Hello! First of all, thank you for your amazing work!

I am here to report what seems to be a bug and I am ready to try to fix it myself if I get enough help. Since I don't have enough knowledge about wine's shenanigans.

## Bug description

Starting from commit 02913f7 of the [wine-staging](https://github.com/wine-staging/wine-staging) code, the Battle.net Launcher from Overwatch doesn't seem to start anymore from Lutris (which enables then to run Overwatch).

I have confirmed at least that if I go to the wine executable from that specific wine-staging version that doesn't work with Lutris and use it to run Battle.net with 
```shell

/wine-staging-02913f7/wine "/path/to/overwatch/drive_c/Program Files (x86)/Battle.net/Battle.net.exe"

```

Battle.net does work: I can login and launch the game (without dxvk)

## Steps to reproduce

- Clone TK-Glitch's git to compile wine-staging with fsync : https://github.com/Frogging-Family/wine-tkg-git
- Open the file ``wine-tkg-git/customization.cfg``
  - Update ``_nomakepkg_dep_resolution_distro`` with the used distro
   - reference wine-staging's commit : ``_staging_version="02913f7"``
- Compile by running `wine-tkg-git/non-makepkg-build.sh`
- Install Overwatch in Lutris on a "regular" wine version (the game won't be downloaded, only Battle.net. It will only take few minutes to complete)
- Verify that the Battle.net Launcher works fine with the regular wine version
- Add the custom compiled wine to Lutris and try use it to run Overwatch

Even though I am using wine-staging with additional patches, I am sure the problem doesn't come from the patchset that is applied on top of wine-staging. Because that same patchset is applied on earlier wine-staging versions and the launcher works fine. But if you wish I can try vanilla wine-staging at that commit and report back.

## Debug information

[lutris-debug-out.txt](https://github.com/lutris/lutris/files/4874204/lutris-debug-out.txt)

[lutris-issue-report.txt](https://github.com/lutris/lutris/files/4874206/lutris-issue-report.txt)

**Comment by AdelKS** [comment]
Hello @strycore, this is *not* a wine Lutris issue so to speak, I am only reporting that if you start including upstream wine-staging. Battle.net launcher seems to not start anymore through Lutris. Even though it does start when ran directly from command line with `wine Battle.net.exe` with that same wine-staging version.

**Comment by AdelKS** [comment]
Hello @tannisroot ! I have recently tested wine-lutris-5.7-8, and it seems that it has gotten the change that makes Battle.net not start anymore. Can someone test it out to see if it's not something specific to my setup ?

**Comment by tannisroot** [comment]
It still works for me.

**Comment by AdelKS** [comment]
Cool thanks! I will explore what wrong.

**Comment by AdelKS** [comment]
Okay I got it, there seem to exist some intricate interaction with different versions of Mesa, MangoHud and Wine.

For those who encountered the same problem: For example disabling Mangohud, using a stable-ish version of Mesa (I tried mesa from git at commit f6aa071), and using the latest wine-tkg protonified version (I am using v5.12) works. The Battle.net launcher and Overwatch do start (then there is some artifacts in overwatch but that's due to latest wine). 

This issue can be closed, since it's not related to Lutris. Thank you for your help!
